### ISSUE TITLE
cli option `-w` is inconsistent and has confusing error messages

### ISSUE THREAD

**Issue by brisvag** [napari/napari#4109]
## 🐛 Bug

While this command works fine, (opening the appropriate plugin widget):

```
napari some_file -w blik
```

This fails:
```
napari -w blik some_file
```
 
```
KeyError: "Plugin 'blik' does not provide any dock widgets"
```

I tested another plugin, which in the second case gave this error instead:
```
KeyError: "Plugin 'napari-properties-plotter' does not provide a widget named '/home/brisvag/tmp/dr_tomo/clean_bin2.mrc'"
```

which is more understandable. So I suppose the bug here is really just in the error message? But how are the two different?

Strangely, this error was a bit hard to reproduce. I was about to write that no plugins worked with:
```
napari -w plugin
```

that is, without any file, but now suddenly they are working again. Can anyone reproduce my issues? And are error messages different depending on the required plugin?

Also, the same error is fired when a non-existing plugin name is passed, which was particularly confusing when I mistakenly passed the *module name* rather than the *plugin name*. These cases should probably say "plugin X does not exist".

PS: I just noticed that `blik` does not figure among the plugins in my `napari --info` output. However, it's definitely installed and working, since the first code block above works fine.

## Expected behavior

Clearer error messages. Also the output seems somehow inconsistent, but it's hard to tell if I was messing up things or it;s just hard to reproduce.

## Environment

```yaml
napari: 0.4.14
Platform: Linux-5.16.9-arch1-1-x86_64-with-glibc2.35
System: Arch Linux
Python: 3.10.2 (main, Jan 15 2022, 19:56:27) [GCC 11.1.0]
Qt: 5.15.2
PyQt5: 5.15.6
NumPy: 1.22.2
SciPy: 1.8.0
Dask: 2022.02.0
VisPy: 0.9.6

OpenGL:
  - GL version:  4.6.0 NVIDIA 510.47.03
  - MAX_TEXTURE_SIZE: 32768

Screens:
  - screen 1: resolution 1920x1080, scale 1.0

Plugins:
  - console: 0.0.4
  - napari-properties-plotter: 0.2.2
  - scikit-image: 0.4.14
  - svg: 0.1.6
```

**Comment by tlambert03** [comment]
> But how are the two different?

They're different because -w is a flag that consumes up to 2 args (plugin name, [widget name])

And in your first one you're opening the path some_file with the widget from plugin blik, but the second one you're opening NO path, but a funny widget name.

As for the error message, just to check, have you merged main today?  There was a PR that just went in to improve the message when a plugin doesn't provide a specific widget. (On mobile, don't remember number).

Lastly, if this is an npe1 plugin run `napari --plugin-info -v` to see any plugin errors that may have prevented the plugin from loading

**Comment by brisvag** [comment]
`main` seems to have fixed. I tried with a clean slate env and installed only the bare minimum, and things make sense now; the difference between error messages was caused by the fact that one of the two plugins was npe1.

Now, there's still a difference, but it's just that the npe2 plugin tells me other widget name options, while npe1 only says that no widgets of that name are available. Consider this closed!
